### PR TITLE
ssd1306: add options to customize the hardware pin layout

### LINKED ISSUE #166: ssd1306 wrong resolution on 128x32 OLED
Using this product on Raspberry Pi Zero W using latest stretch:

Adafruit PiOLED - 128x32 Monochrome OLED Add-on for Raspberry Pi
https://www.adafruit.com/product/3527

The image seems wrong using the `ssd1306` cmd provided in the package:

```
./ssd1306 -h 32
```

Results in:

img_20171023_020257

It only writes to the second half of the screen, with tiny text cut in half.

```
# even though the display has a height of 32, I tried 64 to see
./ssd1306 -h 64
```
Results in:

img_20171023_020332

The display is working fine with their official python library: https://github.com/adafruit/Adafruit_Python_SSD1306/blob/master/Adafruit_SSD1306/SSD1306.py

The **Adafruit 128x64 OLED Bonnet for Raspberry Pi**  is working fine with periph (https://www.adafruit.com/product/3531): 

img_20171023_021111

### PR DESCRIPTION
**Pull request by davidsansome**
Fixes #166

### CODE REVIEW

File: devices/ssd1306/ssd1306.go [hunk 1]
```diff
@@ -65,6 +67,14 @@ type Opts struct {
 	H int
 	// Rotated determines if the display is rotated by 180°.
 	Rotated bool
+	// Sequential corresponds to the Sequential/Alternative COM pin configuration
+	// in the OLED panel hardware. Try toggling this if half the rows appear to be
+	// missing on your display.
+	Sequential bool
+	// SwapTopBottom corresponds to the Left/Right remap COM pin configuration in
+	// the OLD panel hardware. Try toggling this if the top and bottom halves of
```
Comment (maruel) [inline devices/ssd1306/ssd1306.go]: s/OLD/OLED/ :)

**Comment by maruel** [comment]
gohci

**Comment by maruel** [comment]
There are test failures to fix first.

File: devices/ssd1306/ssd1306.go (hunk 1 above)
Comment (davidsansome) [reply devices/ssd1306/ssd1306.go]: Oops, fixed!

**Comment by maruel** [comment]
Thanks a lot for this change. One last request; since this code is already at 100% line coverage, would you mind adding a test case to cover the new line you added that is currently not covered?

**Comment by maruel** [comment]
gohci

**Comment by maruel** [comment]
gohci

**Comment by maruel** [comment]
Thanks a lot for the change, it's really appreciated!

**Pull request merged**